testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPSamlIdPInitiatedVaryingLetterCaseTest.java

```diff
@@ -0,0 +1,305 @@
+        idp.setFirstBrokerLoginFlowAlias(FLOW_AUTO_LINK);
+        final Creator<IdentityProviderResource> idpCreator = Creator.create(testRealm(), idp);
+        
+        IdentityProviderMapperRepresentation samlNameIdMapper = new IdentityProviderMapperRepresentation();
+        samlNameIdMapper.setName("username-nameid-mapper");
+        idpAlias = idp.getAlias();
+        samlNameIdMapper.setIdentityProviderAlias(idpAlias);
+        samlNameIdMapper.setIdentityProviderMapper(UsernameTemplateMapper.PROVIDER_ID);
+        samlNameIdMapper.setConfig(ImmutableMap.<String,String>builder()
+            .put(IdentityProviderMapperModel.SYNC_MODE, "IMPORT")
+            .put(UsernameTemplateMapper.TEMPLATE, "${NAMEID | lowercase}")
+            .put(UsernameTemplateMapper.TARGET, Target.BROKER_ID.name())
+            .build());
+        idpCreator.resource().addMapper(samlNameIdMapper);
+
+        getCleanup().addCleanup(idpCreator);
+    }
+
+    @Before
+    public void setupClients() {
+        getCleanup().addCleanup(Creator.create(testRealm(), ClientBuilder.create()
+          .protocol(SamlProtocol.LOGIN_PROTOCOL)
+          .clientId(EXT_SSO_URL)
+          .baseUrl(EXT_SSO_URL)
+          .attribute(SamlProtocol.SAML_IDP_INITIATED_SSO_URL_NAME, EXT_SSO)
+          .attribute(SamlProtocol.SAML_NAME_ID_FORMAT, JBossSAMLURIConstants.NAMEID_FORMAT_PERSISTENT.get())
+          .attribute(SamlProtocol.SAML_ASSERTION_CONSUMER_URL_POST_ATTRIBUTE, DUMMY_URL)
+          .build())
+        );
+
+        getCleanup().addCleanup(Creator.create(testRealm(), ClientBuilder.create()
+          .clientId(MY_APP)
+          .protocol(OIDCLoginProtocol.LOGIN_PROTOCOL)
+          .baseUrl(oauth.APP_AUTH_ROOT)
+          .build())
+        );
+    }
+
+    @After
+    public void cleanupUsers() {
+        testRealm().userStorage().removeImportedUsers(ldapModelId);
+    }
+
+    @Test
+    @LDAPConnectionParameters(bindType=LDAPConnectionParameters.BindType.SIMPLE, encryption=LDAPConnectionParameters.Encryption.NONE)
+    public void loginLDAPTest() {
+        loginPage.open();
+        loginPage.login(USER_NAME_LDAP, USER_PASSWORD);
+        appPage.assertCurrent();
+        Assert.assertEquals(AppPage.RequestType.AUTH_RESPONSE, appPage.getRequestType());
+        Assert.assertNotNull(oauth.getCurrentQuery().get(OAuth2Constants.CODE));
+        appPage.logout();
+    }
+
+    protected URI getAuthServerBrokerSamlEndpoint(String realm, String identityProviderAlias, String samlClientId) throws IllegalArgumentException, UriBuilderException {
+        return RealmsResource
+                .realmBaseUrl(UriBuilder.fromUri(getAuthServerRoot()))
+                .path("broker/{idp-name}/endpoint/clients/{client-id}")
+                .build(realm, identityProviderAlias, samlClientId);
+    }
+
+    @Test
+    @LDAPConnectionParameters(bindType=LDAPConnectionParameters.BindType.SIMPLE, encryption=LDAPConnectionParameters.Encryption.NONE)
```

Comment:
I think the annotation @LDAPConnectionParameters is not necessary and probably should not be used in this test. Or is there any reason for using and not rely on the default?

If the annotation is needed for this test, it will be good to run the LDAP tests in pipeline and check if this works for other LDAP servers besides ApacheDS, as I have some doubts it will work. The only test where this annotation is currently used is LDAPUserLoginTest, which is supposed to execute just on LDAP embedded ApacheDS server.  However not sure if LDAP tests in pipeline are not currently broken :(

Same comment applies for all other occurences of this annotation.